Release new DAMs (#95)
CloudCannon/platform-documentation: articles/creating-a-digitalocean-spaces-dam.mdx

### Configure CORS policies

You will need to configure your CORS policies to allow CloudCannon to interact with your bucket. In the Settings tab for your bucket in DigitalOcean, click the Add button next to "CORS Configurations" to add a new policy.

Minimally, you need to allow the GET and PUT methods, and the Content-Type header for https://app.cloudcannon.com.

<comp.DocsImage path="ye_olde_images/digitalocean-spaces-cors-config.png" alt="Screenshot of the CORS policy editor in DigitalOcean" type="screenshot" />

### Create an API token

You will also need to create an API token to allow CloudCannon to access your bucket. Click on "API" in the sidebar in DigitalOcean, and open the "Spaces Keys" tab.

Click "Generate New Key" and type any name to create the key. Save the secret token somewhere safe, and note down the Key for later.

<comp.DocsImage path="ye_olde_images/digitalocean-spaces-access-key.png" alt="Screenshot of the menu in DigitalOcean for creating API keys" type="screenshot" />

### Make your bucket items public

> By default, all the objects in your DigitalOcean Spaces bucket are private, and cannot be viewed without authentication. You will need to make them public, so that they can be viewed on your site and within CloudCannon. You can configure the permissions on each object separately within DigitalOcean, but ideally we want to make all objects in the bucket public default.

Reviewer: Missing "by" in the very last sentence, should read "public by default"

> DigitalOcean Spaces is built on top of Amazon's S3 service, so you can use the AWS CLI to make changes to your bucket policy. [See the AWS guide for installing the CLI here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).

Reviewer 2: ```suggestion
DigitalOcean Spaces is compatible with Amazon's S3 API, so you can use the AWS CLI to make changes to your bucket policy. [See the AWS guide for installing the CLI here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
```

First, make a file called `policy.json` with the following contents, substituting `<YOUR-BUCKET-NAME>` with the name of your bucket. This policy will allow public read access for all objects within your bucket.

<comp.CodeBlock language="json" source="policy.json">
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "PublicReadGetObject",
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::<YOUR-BUCKET-NAME>/*"
            ]
        }
    ]
}
```
</comp.CodeBlock>

Now run this command in your terminal, using the `aws` command line tool. Again, you will need to substitute the angle-bracketed placeholders with data from DigitalOcean.

<comp.CodeBlock language="bash">
```
AWS_ACCESS_KEY_ID=<YOUR-API-TOKEN-KEY> \
AWS_SECRET_ACCESS_KEY=<YOUR-API-TOKEN-SECRET> \
aws s3api put-bucket-policy \
    --bucket=<YOUR-BUCKET-NAME> \
    --policy=file://./policy.json \
    --endpoint-url=<YOUR-BUCKET-ENDPOINT>
```
</comp.CodeBlock>

### Connect your DAM to CloudCannon

> You should now have everything you need to connect CloudCannon with your R2 DAM.

Reviewer: 🚨🚨🚨 REFERENCES TO CLOUDFLARE R2 IN THE DIGITAL OCEAN DOC 🚨🚨🚨
